src/internal/agent/start.go

```diff
@@ -28,20 +33,48 @@ const (
 )
 
 // StartWebhook launches the Zarf agent mutating webhook in the cluster.
-func StartWebhook(ctx context.Context) error {
-	srv, err := agentHttp.NewAdmissionServer(ctx, httpPort)
-	if err != nil {
-		return err
-	}
-	return startServer(ctx, srv)
+func StartWebhook(ctx context.Context, cluster *cluster.Cluster) error {
```

Comment:
It could be pretty easy to add tests for this with an test http client.

Reply:
I agree, I am going to revisit this package in the future to get more tests in.